On The Map/OTMClient.swift

```diff
@@ -0,0 +1,192 @@
+        }
+        
+        do {
+            request.HTTPBody = try NSJSONSerialization.dataWithJSONObject(jsonBody, options: [])
+        } catch _ as NSError {
+            request.HTTPBody = nil
+        }
+        
+    
+        /* 4. Make the request */
+        let task = session.dataTaskWithRequest(request) { (data, response, error) in
+            
+            if let JSONError = error {
+                
+                _ = OTMClient.errorForData(data, response: response, error: JSONError)
+                print(error)
+                completionHandlerForPOST(result: nil, error: error)
+            } else {
+                var newData = data
+                if(udacity){
+                    newData = data!.subdataWithRange(NSMakeRange(5, data!.length - 5)) 
+                }
+                /* 5/6. Parse the data and use the data (happens in completion handler) */
+                self.convertDataWithCompletionHandler(newData!, completionHandlerForConvertData: completionHandlerForPOST)
+            }
+            
+        }
+        
+        /* 7. Start the request */
+        task.resume()
+        
+        return task
+    }
+    
+    // MARK: - GET
+    
+    func taskForGETMethod(method: String, udacity: Bool, parameters: [String:AnyObject]?, completionHandlerForGET: (result: AnyObject!, error: NSError?) -> Void) -> NSURLSessionDataTask {
+        
+        var urlString: String
+        
+        if let mutableParameters = parameters {
+            urlString = method + OTMClient.escapedParameters(mutableParameters)
+        } else {
+            urlString = method
+        }
+        
+        let request = NSMutableURLRequest(URL: NSURL(string: urlString)!)
+        
+        if (!udacity) {
+            request.addValue("QrX47CA9cyuGewLdsL7o5Eb8iug6Em8ye0dnAbIr", forHTTPHeaderField: "X-Parse-Application-Id")
+            request.addValue("QuWThTdiRmTux3YaDseUSEpUKo7aBYM737yKd4gY", forHTTPHeaderField: "X-Parse-REST-API-Key")
+
+        }
+        
+        let task = session.dataTaskWithRequest(request) { (data, response, error) in
+            if let dataError = error {
+                OTMClient.errorForData(data, response: response, error: dataError)
+                completionHandlerForGET(result: nil, error: dataError)
+            } else {
+                var newData = data
+                if(udacity){
+                    newData = data!.subdataWithRange(NSMakeRange(5, data!.length - 5)) /* subset response data! */
+                }
+                self.convertDataWithCompletionHandler(newData!, completionHandlerForConvertData: completionHandlerForGET)
+            }
+        }
+        task.resume()
+        
+        return task
+    }
+    
+    // given raw JSON, return a usable Foundation object
+    private func convertDataWithCompletionHandler(data: NSData, completionHandlerForConvertData: (result: AnyObject!, error: NSError?) -> Void) {
```

Comment:
Can this be a declared as a `class func` instead of a `private func` ?
That way, you can declare `taskForPOSTMethod` and `taskForGETMethod` as `class func` too, and get rid of your Singleton nonsense below.